v3iofs/fs.py

```diff
@@ -52,29 +58,79 @@ def __init__(self, v3io_api=None, v3io_access_key=None, **kw):
             transport_kind='requests',
         )
 
+    def _details(self, contents, **kwargs):
+        pathlist = []
+        for c in contents:
+            data = {}
+            data['name'] = c['name'].lstrip("/")
+            if c['size'] is not None:
+                data['type'] = 'file'
+                data['size'] = c['size']
+            else:
+                data['type'] = 'directory'
+                data['size']= 0
+            pathlist.append(data)
+        return pathlist
+
     def ls(self, path, detail=True, **kwargs):
         """Lists files & directories under path"""
+        
         container, path = split_container(path)
-
-        if not container:
+        if container == "":
```

Comment:
Why?

Reply:
If passing fs.ls("/"), expect to return a list of top level directories.

Reply:
I was asking about the change from `if not container` to `if contanier != ""`

Reply:
A user erroneously passing a path such as:  " /xzy.parquet" would return container = True.  This a) seemed more explicit, particularly when looking at the response object that would be returned in the original code, and b) when paired with the updated code changes, returns an explicit error to the user on the issue with their request.